webscale/tls/aliaskeyedcache_test.go

```diff
@@ -0,0 +1,266 @@
+	}
+}
+
+func TestNewAliasKeyedCache_set_defaultCertificate_to_nil(t *testing.T) {
+	cache := NewAliasKeyedCache()
+	if cache.defaultCertificate != nil {
+		t.Fatalf("Expected an initialized cache defaultCertificate to be nil, got %+v", cache)
+	}
+}
+
+func TestLoad_returns_error_given_invalid_file(t *testing.T) {
+	cert, err := os.Create("cert1.pem")
+	if err != nil {
+		t.Fatalf("Unexpected error %s", err)
+	}
+	defer os.Remove(cert.Name())
+	content :=
+		`-----BEGIN RSA PRIVATE KEY-----
+MIICXQIBAAKBgQChxPY4u7R8EMpQjMSwkKrnEd3wLGaz9Zm2HmQi2x
+-----END RSA PRIVATE KEY-----
+-----BEGIN CERTIFICATE-----
+MIICLjCCAZcCAQEwDQYJKoZIhvcNAQEFBQAwWzELMAkGA1UEBhMCVVMxDTALBgNV
+-----END CERTIFICATE-----
+}`
+	cert.Write([]byte(content))
+
+	cache := NewAliasKeyedCache()
+	err = cache.Load(cert.Name(), []string{"example.com"})
+	if err == nil {
+		t.Fatalf("Expect an error due to an invalid PEM file")
+	}
+}
+
+func TestLoad_stores_certificate_for_aliases(t *testing.T) {
+	cert, err := os.Create("cert1.pem")
+	if err != nil {
+		t.Fatalf("Unexpected error %s", err)
+	}
+	defer os.Remove(cert.Name())
+	content :=
+		`-----BEGIN RSA PRIVATE KEY-----
+MIICXQIBAAKBgQChxPYdRkkmetH2K4u7R8EMpQjMSwkKrnEd3wLGaz9Zm2HmQi2x
+wqYgi2i4bCsZp2biWapH9uD+gpmcYfnDb1Fk5CbdS4ZAKwVUWU/Los4FOvxXtK+6
+hEcPRtpQJ95xa27vqO6qFzG2ez0f1jx153IyYwJ/WtlawUTwTxhUkiGbSQIDAQAB
+AoGABgXXQ+/B+XTJLGkioq5hOZ9LXI/Onl8wRvRungSQLz3hvzjnip68oKmQFI2y
+bRoWcob0GAnRBqjGH1RmgCg8132f0PfWXD+xMKwEM2ut8PeEbW8b98KEswgud22K
+q8hJBERvB0WSC+h+N+IGJBK8d8AvUlyVeOjIUk0Xs7PLMIECQQDMBwTKh1GwZWo1
+RYoK6Wfr6g9xkhlXEQ+wmzn0uDrIxPP6NICKt7SFLTiRshQVFcZ+wwjCbVGdK9Cy
+XoZNitCdAkEAyvo5EJgaVEiVqPwL40qG6ci9ZjibdcQaxe6M91XqyMU0L6PzWSNm
+aoIDx6+DTbAYTvEjFy4v7xcfC08gKCrnnQJBAKMqyc4ewlnMAVBxOKDZYV7uZUNy
+kAltf5rByWvJGloOCQCElHhbymbnb2I1hJIIRCKEX7D+NFL6A4Fizw2cgpECQQCB
+mMoeoj8NWVrVDji44rjJQ/ZJ8hKwWomNnwY6VY0Wq3LqiA+z9jpJ/sFTGekIDUs3
+/Bafkkngqi6UFe0+OEaxAkA5GnIgnQaIToj28vpax//p1AnA4i/te1cYKaJzBZpy
+lP41So/Dpf1UIW6GmZo2w4PVKj3hc6/FUegVLmQxZffD
+-----END RSA PRIVATE KEY-----
+-----BEGIN CERTIFICATE-----
+MIICLjCCAZcCAQEwDQYJKoZIhvcNAQEFBQAwWzELMAkGA1UEBhMCVVMxDTALBgNV
+BAgMBE9oaW8xETAPBgNVBAcMCENvbHVtYnVzMRAwDgYDVQQKDAdNeUNlcnRzMRgw
+FgYDVQQDDA93d3cubXljZXJ0cy5jb20wHhcNMjEwNDIzMjIwMDM4WhcNMjIwNDIz
+MjIwMDM4WjBkMQswCQYDVQQGEwJVUzERMA8GA1UECAwIQ29sb3JhZG8xEDAOBgNV
+BAcMB0JvdWxkZXIxETAPBgNVBAoMCFdlYnNjYWxlMR0wGwYDVQQDDBQ3NTkxKi5s
+YWdyYW5nZS5uaW5qYTCBnzANBgkqhkiG9w0BAQEFAAOBjQAwgYkCgYEAocT2HUZJ
+JnrR9iuLu0fBDKUIzEsJCq5xHd8Cxms/WZth5kItscKmIItouGwrGadm4lmqR/bg
+/oKZnGH5w29RZOQm3UuGQCsFVFlPy6LOBTr8V7SvuoRHD0baUCfecWtu76juqhcx
+tns9H9Y8dedyMmMCf1rZWsFE8E8YVJIhm0kCAwEAATANBgkqhkiG9w0BAQUFAAOB
+gQCV0nzUudq58dwZBbVG0U8dIQ+Fb4f/WUYcr9F3q3o4A4tXN5xGXRqbtqyERT/J
+dnJycx1mp+X6hiTXJaajhEy+ljhz+ubJmIrpRUh+fhxboI5ml5O1J+hdrFP2He0/
+9AwI9B05dYBoizG3WuXll8ctbQE1eDi70GzSyQ5gflpyDw==
+-----END CERTIFICATE-----
+	`
+	cert.Write([]byte(content))
+	cache := NewAliasKeyedCache()
```

Comment:
Might be useful to write a helper method to create a cert file given the contents. You have at least 4 places in the test code where you are creating a cert. You can probably use it in the tests for `certificates.go` too.

Reply:
Added test PEM blocks for a private key and a certificate and created a `createTestCertificateOnDisk` function.